app/lib/task/backend.dart

```diff
@@ -427,7 +428,7 @@ class TaskBackend {
               attempts: 0,
             ),
         });
-      state.derivePendingAt();
+      state.derivePendingAt(isHighPriority: isHighPriority);
```

Comment:
```suggestion
      state.pendingAt = initialTimestamp;
```

Reply:
We have an initial value, and there is a comment  for why it is so.

I'm not sure we ever set `pendingAt` to it anywhere.. so it might be fine to do this.

Reply:
Updated it inside the `derivePendingAt`. Maybe it is not the best place for it, but then we'd need to `if (highPrio) {setToZero()} else {derivePendingAt()}` for the two places in this pr, and I think it has better read-flow this way.

Reply:
I still don't believe this will work.

```dart
      // Stop transaction, if there is no changes to be made!
      if (untrackedVersions.isEmpty && deselectedVersions.isEmpty) {
        return false;
      }
```

will probably cause problems...

Why not just: call `trackPackage(packageName)` then load `PackageState` and set `pendingAt = initialTimestamp`.
It's a bit cleaner than moving logic into a function that's now going to be half broken, because we probably can't keep track of all the concerns it has to deal with.